kernel/aster-nix/src/time/clock.rs

```diff
@@ -0,0 +1,287 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use alloc::{boxed::Box, sync::Arc};
+use core::time::Duration;
+
+use aster_frame::{
+    arch::{jiffies_as_duration, register_interrupt_callback, TIMER_FREQ},
+    cpu_local,
+    sync::SpinLock,
+    CpuLocal,
+};
+use aster_time::{read_monotonic_time, NANOS_PER_SECOND};
+use int_to_c_enum::TryFromInt;
+use paste::paste;
+use spin::Once;
+
+use super::{system_time::START_TIME_AS_DURATION, timer::TimerManager, SystemTime};
+
+type Nanos = u64;
+
+#[derive(Debug, Copy, Clone, TryFromInt, PartialEq)]
+#[repr(i32)]
+pub enum ClockID {
+    CLOCK_REALTIME = 0,
+    CLOCK_MONOTONIC = 1,
+    CLOCK_PROCESS_CPUTIME_ID = 2,
+    CLOCK_THREAD_CPUTIME_ID = 3,
+    CLOCK_MONOTONIC_RAW = 4,
+    CLOCK_REALTIME_COARSE = 5,
+    CLOCK_MONOTONIC_COARSE = 6,
+    CLOCK_BOOTTIME = 7,
+}
+
+/// A trait that can abstract clocks which have the ability to read time,
+/// and has a fixed resolution.
+pub trait Clock: Send + Sync {
+    /// Read the current time of this clock.
+    fn read_time(&self) -> Duration;
+
+    /// The resolution of this clock.
+    /// Set to the resolution of system time interrupt by default.
+    fn resolution(&self) -> Nanos
+    where
+        Self: Sized,
+    {
+        NANOS_PER_SECOND as u64 / TIMER_FREQ
+    }
+}
+
+/// The Clock that reads the jiffies, and turn the counter into `Duration`.
+#[derive(Default)]
+pub struct JiffiesClock;
+
+/// `RealTimeClock` represents a clock that provides the current real time.
+#[derive(Default)]
+pub struct RealTimeClock;
+
+/// `MonotonicClock` represents a clock that measures time in a way that is
+/// monotonically increasing since the system was booted.
+#[derive(Default)]
+pub struct MonotonicClock;
+
+/// `RealTimeCoarseClock` is a coarse-grained version of a real-time clock.
+#[derive(Default)]
+pub struct RealTimeCoarseClock;
```

Comment:
Thanks for your commemt. I will supplement the docs. BTW, I want to classify the misunderstanding that `RealTimeCoarseClock` is also a global clock. It just does not have a global `TimerManager`. This is because people rarely use coarse's time as a base for timer. I will supplement these relevant explanation in the doc.